R/check.R

```diff
@@ -671,3 +671,43 @@ check_observation_indicator <- function(
   }
   return(invisible(NULL))
 }
+
+#' Check design matrix sparsity
+#'
+#' This function checks the sparsity of a design matrix and provides a
+#' recommendation if the matrix is considered sparse.
+#'
+#' @param matrix A numeric matrix to be checked for sparsity.
+#' @param sparsity_threshold A numeric value between 0 and 1 indicating the
+#' threshold for considering a matrix sparse. Default is 0.9.
+#' @param min_matrix_size An integer indicating the minimum size of the matrix
+#' for which to perform the sparsity check. Default is 50.
+#' @param name A character string specifying the name of the design
+#' matrix. Default is "checked".
+#'
+#' @return This function is used for its side effect of providing an
```

Comment:
seems like this should return TRUE or FALSE? i get the use is apparently for the message, but a bit weird to ask a yes-no question and not return the answer. Is there a way for me to programmatically say "if this check indicates sparse, set sprase_design = TRUE"?

Reply:
I don't disagreee but this does fit with the rest of the check funcitons that return NULL